src/instructlab/sdg/generate_data.py

```diff
@@ -220,6 +220,11 @@ def _sdg_init(ctx, pipeline):
     data_dirs = [os.path.join(xdg_data_home(), "instructlab", "sdg")]
     data_dirs.extend(os.path.join(dir, "instructlab", "sdg") for dir in xdg_data_dirs())
 
+    # Initialize docling model path
+    docling_model_path = os.path.join(xdg_data_home(), "models", "docling")
+    # Ensure the `docling_model_path` directory exists
+    os.makedirs(docling_model_path, exist_ok=True)
+
```

Comment:
a few suggestions:
- the models themselves will be in `/usr/share/instructlab/sdg/sdg-models/docling-models`
- we should check for the models directory across all dirs populated in `data_dirs`
- we don't need to make sure the directory exists. If it does we proceed, if it does not then we can just set the docling_model_path to be `None` and let it get handled later on 
- what we really need to do is look for a `config.yaml` at `/usr/share/instructlab/models/sdg-models` that will contain the path where the actual models are stored. That's the path we need to pass ahead if it exists in the file

something along the lines of:

```suggestion
    sdg_models_path = docling_models_path = None 
    for d in data_dirs:
        if os.path.exists(os.path.join(d, "models")):
          sdg_models_path = os.path.join(d, "models")
          break
     
     if sdg_models_path is not None:
       try:
          with open(os.path.join(sdg_models_path, "config.yaml"), "r", encoding="utf-8") as file:
              config = yaml.safe_load(file)
              docling_models_path = config['models'][0]['path']
        except (FileNotFoundError, NotADirectoryError, PermissionsError) as e:
              log.warning(f"unable to read docling models path from config.yaml")
```

so we know after this point `docling_models_path` is either set correctly, or it is `None`